docs/source/developer/internationalization.rst

```diff
@@ -0,0 +1,126 @@
+  defines the Python package template of a language pack.
+
+The *language-packs* repository is the main entry point. It interacts with Crowdin to publish
+newer source strings and get the latest translation. It also creates and updates the language packs.
+And finally it publish them. All those actions are carried out using helpers defined in 
+``jupyterlab-translate`` and the cookiecutter template.
+
+Workflows
+---------
+
+The workflows at play will be described next. In the order they are usually called.
+
+.. note::
+
+    Automatic tasks are carried out through the *jupyterlab-bot*. To do that, that bot has
+    access to the GitHub repository, the Crowdin project and all language packs projects on
+    `PyPI <https://www.pypi.org>`_.
+
+Source strings generation
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Source strings are extracted for JupyterLab and a list of extensions at a given version defined
+in `repository-map.yaml <https://github.com/jupyterlab/language-packs/blob/master/repository-map.yml>`_
+file in *language-packs* repository. The workflow to trigger an update is as follow:
+
+1. Edit ``repository-map.yaml`` by adding new repositories and/or updating the target version.
+2. Push the change in a pull request.
+3. Once the pull request is merged, the workflow `Update source strings <https://github.com/jupyterlab/language-packs/blob/master/.github/workflows/update_pot.yml>`_ will automatically be triggered.
+4. That workflow will open a new pull request that update the source strings and optionally the Crowdin configuration.
+5. Once that pul request is merged, Crowdin will upload the new source strings automatically.
+
+.. note::
+
+    Crowdin is uploading automatically its source strings using `GitHub Integration <https://support.crowdin.com/github-integration/>`_ set up
+    with the Crowdin account of *jupyterlab-bot*.
+
+    The script used for this workflow is `02_update_catalogs.py <https://github.com/jupyterlab/language-packs/blob/master/scripts/02_update_catalogs.py>`_.
+
+Translation update
+^^^^^^^^^^^^^^^^^^
+
+The new and/or updated translation are automatically pushed to the *language-packs* repository.
+The workflow is as follow:
+
+1. A contributor updates the translation on JupyterLab Crowdin project
+2. A new commit with those changes are pushed to the *language-packs* repository on a branch named
+   ``l10n_master``.
+3. If there is no pull request associated with that branch, a new pull request will be opened.
+4. A maintainer needs to merge that pull request.
+
+.. note::
+
+    Crowdin is uploading automatically the translation using `GitHub Integration <https://support.crowdin.com/github-integration/>`_ set up
+    with the Crowdin account of *jupyterlab-bot*. Hence the commits and pull request is attributed
+    to the bot.
+
+    If the branch is deleted, it will be re-created.
+
+.. warning::
+
+    To avoid merge conflicts on those translation update pull requests, they must to be merged before
```

Comment:
Maybe we could add, that in case of conflicts on a PR we simply can close the PR and wait for crowdin to re-create a new PR that will no longer have conflicts?

Reply:
Done